### PR TITLE
Add move method to automation rule

### PR DESCRIPTION
**Pull request by stsewd**
This is needed to re-organize rules
without explicit changing all priorities
or changing one by one

I just need to add tests for negative steps, and add a comment why just doing an `.update(F('priority') + 1)` doesn't work

### CODE REVIEW

**Comment by stsewd** [comment]
Oops, I use the manager introduced in #5995, that PR should be merged first

**Comment by humitos** [comment]
What does it mean to "move an automation rule N steps"?

**Comment by stsewd** [comment]
@humitos if you have this rules

1) One
2) Two
3) Three

The automation rules are run in that order (according to their priority). User can rearrange the rules to change the order they are run. Like

1) Three
2) Two
3) One

Users shouldn't need to update each priority version one by one. Think of it like a drag and drop thing

**Comment by stsewd** [comment]
> I feel this implementation is too complex and does not manage moving one step up (like steps=-1, at least in the tests).

It does, tests are still missing. It's on the description.

> move that Automation Rule where it should be in the list (steps) set the index of each Automation Rule in the list as priority

We can't save the values like that, it breaks the constraint of unique (project, priority)

> So, we will end up always with 0, 1, 2, 3 as priority if we have 4 rules, instead of dealing with weird priority numbers.

Current implementation always keeps the range, it doesn't add weird numbers

**Comment by humitos** [comment]
> We can't save the values like that, it breaks the constraint of unique (project, priority)

Why it will break that constraint if we will have priorities like 0, 1, 2, 3, 4 tied to one project?

> Current implementation always keeps the range, it doesn't add weird numbers

This is the weird number that I was reffering to: https://github.com/readthedocs/readthedocs.org/pull/5998/files#diff-763de362f34a86b264903b372e7dd27fR1012

**Comment by stsewd** [comment]
> This is the weird number that I was reffering to

That's is temporary, to avoid the unique constraint

> Why it will break that constraint if we will have priorities like 0, 1, 2, 3, 4 tied to one project?

In the final result not, but in the intermediate steps, when you try to save or update an object, other object will have the same priority. That's why I can't use just one `update` in the current code and  need to iterate over each object.

**Comment by stsewd** [comment]
This is ready for review. This is on top of #5995 (that should be merged first)

File: readthedocs/builds/managers.py [hunk 1]
```diff
@@ -179,3 +179,37 @@ class InternalBuildManager(SettingsOverrideObject):
 
 class ExternalBuildManager(SettingsOverrideObject):
     _default_class = ExternalBuildManagerBase
+
+
+class VersionAutomationRuleManager(PolymorphicManager):
+
+    def append_rule(
```
Comment (humitos) [inline readthedocs/builds/managers.py]: nit: I'd use `add_rule` to keep consistency with other methods that we already have, like `add_subproject`.

Comment (stsewd) [reply readthedocs/builds/managers.py]: This is part of #5995

**Comment by stsewd** [comment]
I also overrode the delete method to keep the priority numbers the same after delete

**Comment by ericholscher** [comment]
I like having the DB restriction though. I don't trust code not to mess it up, and then we'd end up with weird bugs. This will lead to different weird bugs, but at least these weird bugs will be understandable :)
